pkg/controller/database/rdsinstance.go

```diff
@@ -89,11 +89,29 @@ type external struct {
 	kube   client.Client
 }
 
-func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
 	cr, ok := mg.(*v1beta1.RDSInstance)
 	if !ok {
 		return managed.ExternalObservation{}, errors.New(errNotRDSInstance)
 	}
+
+	pwdChanged := false
```

Comment:
I believe this check belongs to `IsUpToDate` function. As you said, we don't have another resource whose `IsUpToDate` func has a client but we don't have another resource in AWS that takes a secret ref as input either, so this could be example to future implementations.

I would suggest implementing this functionality in a way to be called like `pw, changed, err := rds.CheckPassword(...)` so that we can call it in `IsUpToDate`, `Update` and `Create`, essentially wherever we need the password. `pkg/clients/rds/rds.go` could be a good fit for this utility.